docs/reference/projectyml.rst

```diff
@@ -50,66 +50,174 @@ The code snippet below provides an example of a complete ``project.yml`` file:
-        - <url of a python package repository>
+        index-url: https://my_repo.secure.company.com/repository/inmanta-production
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
+        pre: false
+        use-system-config: false
+
+.. note::
+    The ``pre`` and ``index-url`` options defined in the ``project.yml`` pip section will always take precedence over
+    the corresponding pip options, even when ``use-system-config`` is set to true.
+
+    The ``extra-index-url`` option will extend the extra indexes already set at the system level (if any).
+
+    Other pip-related environment variables are not overridden.
+
+    To use the system's pip configuration without altering it, leave the corresponding option unset.
 
 
 .. note::
-    The pip config file can also be used in combination with ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file:
 
-    * If the pip config is used (by setting ``use_config_file`` to ``true``), the ``index-url`` specified in the pip config file will take precedence and the ``index-urls`` specified in the ``pip`` section of the ``project.yml`` file will be used as ``extra-index-urls`` when installing with pip.
-    * If the pip config is not used (by setting ``use_config_file`` to ``False``), then the first ``index_url`` specified in the project.yml will be used as an ``index_url`` and all the following ones will be used as ``extra-index-urls`` when installing with pip.
+    Set up authentication towards the index using netrc. See this :ref:`section<setting_up_pip_index_authentication>`
+    for more information.
+
+.. _migrate_to_project_wide_pip_config:
+
+Migrate to project-wide pip config
+----------------------------------
+
+``inmanta-core 11.0.0`` introduced new options to configure pip settings for the whole project in a centralized way.
+For detailed information, see :ref:`here<specify_location_pip>`. The following code sample can be used as a baseline
+in the ``project.yml`` file:
+
+
+.. code-block:: yaml
+
+    pip:
+        index-url: https://my_repo.secure.company.com/repository/inmanta-production
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
+        pre: false
+        use-system-config: false
+
+When upgrading to ``inmanta-service-orchestrator 7.0.0`` or ``inmanta 2024.0``, this section can be used as
+a migration guide.
+
+
+All the v2 module sources currently set in a ``repo`` section of the ``project_yml`` with type ``package`` should
+also be duplicated in the ``pip.index-url`` (and ``pip.extra-index-url`` if more than one index is being used).
+
+If you want to allow pre-releases for v2 modules and other Python packages, set ``pip.pre = true`` in the project config
+file. This used to be controlled by the :class:`~inmanta.module.InstallMode` set at the project level or at a module
+level.
+
+Make sure the agents have access to the index(es) configured at the project level.
```

Comment:
Should I remove this too ? Since we're not mentioning non-autostarted agents ?

Reply:
@sanderr wdyt ?

Reply:
@wouterdb do you have an opinion?

Reply:
It's a detail, correct, obvious but important, I have no strong opinion so I would leave it in.